### PR TITLE
Privileged app passwords

### PR DESCRIPTION
**Pull request by dholms**
Adds in a notion of a "privileged" app password which has greater access than a normal app password - for instance to sensitive account state such as chats - but still less access than a full access token.

### CODE REVIEW

File: packages/pds/src/account-manager/db/schema/app-password.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ export interface AppPassword {
   name: string
   passwordScrypt: string
   createdAt: string
+  privileged: 0 | 1 | null
```
Comment (devinivy) [inline packages/pds/src/account-manager/db/schema/app-password.ts:6]: What does the `null` do for us?

Comment (dholms) [reply packages/pds/src/account-manager/db/schema/app-password.ts:6]: I don't think you can make new "non-nullable" columns in sqlite unfortunately so we're just stuck with the `null` as a result of the migration :/

Comment (dholms) [reply packages/pds/src/account-manager/db/schema/app-password.ts:6]: (may be missing some sqlite feature that allows this, but that was the case the last time i checked)

File: packages/pds/tests/app-passwords.test.ts [hunk 2]
```diff
@@ -90,7 +138,51 @@ describe('app_passwords', () => {
       },
     )
 
-    const attempt = appAgent.api.com.atproto.server.createAppPassword(
+    const priviAttempt = appAgent.api.com.atproto.server.getServiceAuth({
+      aud: 'did:example:test',
+    })
+    await expect(priviAttempt).rejects.toThrow('Bad token scope')
```
Comment (devinivy) [inline packages/pds/tests/app-passwords.test.ts:144]: I wonder if these lines were intended to land in a different test.

File: packages/pds/src/account-manager/db/schema/app-password.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/account-manager/db/schema/app-password.ts:6]: Huh, I really thought you could!  If you can, you would def need to set a default at the same time.

Comment (devinivy) [reply packages/pds/src/account-manager/db/schema/app-password.ts:6]: e.g.
```sql
ALTER TABLE app_password
  ADD COLUMN privileged NOT NULL DEFAULT 0;
```

**Review by devinivy — APPROVED** [review]

File: packages/pds/src/account-manager/db/schema/app-password.ts (hunk 1 above)
Comment (dholms) [reply packages/pds/src/account-manager/db/schema/app-password.ts:6]: oh whatdya know, you can as long as you supply a default. good catch 👍

File: packages/pds/tests/app-passwords.test.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/tests/app-passwords.test.ts:144]: ah nope - that's just an example route that allows only privileged app password auth

i added some comments to clarify 👌

**Pull request merged**